### ISSUE TITLE
Not working after installing R2 release of the RelaSense SDK

### ISSUE THREAD

**Issue by kosowski** [libgdx/gdx-realsense#10]
After updating the SDK to the R2 release (available at http://makebettercode.com/realsensesdk/en) the library is not working. The call to PXCSenseManager.CreateInstance() returns always null.

It was working great with the previous release. Any plans to upgrade? Thanks!

**Comment by faluck** [comment]
I am experiencing the same problem

**Comment by faluck** [comment]
does anyone still have the installer of the previous SDK?

**Comment by badlogic** [comment]
Sorry, folks, i'm currently extremely busy with work and won't have time to
look into this for the next few weeks. If anyone wants to try and fix it i
may be able to give some guidance. The build instructions are in the
README.md.

**Comment by faluck** [comment]
was reference to the readme the guidance you mentioned?

**Comment by badlogic** [comment]
Guidance is me answering any questions that come up :)

**Comment by faluck** [comment]
i suppose the underlying problem kosowski and I are facing are somehow of similar nature

senseManager = PXCSenseManager.CreateInstance();
        senseManager.EnableStream(StreamType.STREAM_TYPE_DEPTH);
        senseManager.EnableHand();
        handModule = senseManager.QueryHand();
        senseManager.Init();

```
        handData = handModule.CreateOutput(); // <- this returns null and crashes the program
```

running the camera without handData works fine though

**Comment by badlogic** [comment]
Could you consult the latest real sense SDK docs and see if the
requirements regarding streams etc. changed? Also, is handModule null or is
handData null? From your description it appears that hand module is
actually null and the line you highlight crashes.

**Comment by faluck** [comment]
you're right, handModule is indeed null

System.out.println(senseManager.EnableHand());

gives me

PXC_STATUS_ITEM_UNAVAILABLE

**Comment by faluck** [comment]
oh and ill check the docs now

**Comment by badlogic** [comment]
In that case I'd check with the SDK 2 docs if anything changed regarding
the initialization. Best case: there's a change log detailing the changes.
Worst case: you have to check the new docs.

**Comment by faluck** [comment]
okay i found the doc (https://software.intel.com/sites/default/files/managed/af/a1/realsense-sdk-gold-release-notes.pdf)

under "Frameworks Support (C#, Unity, Web support, Java and Processing):
Issue" it says:

Advanced streaming through the
PXCMCaptureManager interface does not
work as designed. The function
LocateStreams may return failed on valid
stream requests.

Avoid using the LocateStreams function. Use
PXCMSenseManager for color/depth streaming.

and under "Hand Tracking and Gesture Recognition:" it says:

In a scenario when one application is setting
the camera profile (resolution, FPS etc.) to an
unsupported configuration by Hands, accuracy
might be impacted
Ensure camera profile is set to one of the
supported Hands configurations

other than that i couldn't find anything relevant

**Comment by badlogic** [comment]
Can you try the samples that come with the SDK? Do those work?

**Comment by faluck** [comment]
![capture](https://cloud.githubusercontent.com/assets/11477086/7104947/533f7976-e101-11e4-99e2-8df90e145e9f.PNG)

**Comment by faluck** [comment]
that works, the light on the realsense camera briefly goes when i try to run the code but could there be some interference with my laptop camera?

**Comment by Morgam** [comment]
I have the same problem as kosowski, that is, PXCSenseManager.CreateInstance() returns always null. I also tried PXCSession.CreateInstance(), which returns null as well. I assume that this is different issue than faluck is experiencing as I can get senseManager instance in the first place (and the camera doesn’t seem to switch on). I did as told in https://github.com/libgdx/gdx-realsense in setting up the project. I first tried gdx-1.5.5. and subsequently versions 1.5.0 and 1.4.1, and had the same problem.

I have Intel RealSense SDK 4.0.0.52526 (I haven’t tried earlier versions). Camera and SDK examples (in C#, Java and C++) work, as well as SDK Capture Viewer.

In my view, there is something broken in libGDX RealSense SDK Integration, which I would very much get working. Thank you for starting this project!!

**Comment by Morgam** [comment]
Did anyone solve the issue kosowski and I are having? 

 I would really appreciate any help you could give...

**Comment by faluck** [comment]
any updates?

**Comment by badlogic** [comment]
I'm afraid not. As i said, you may try to build the bindings against SDK 2
following the instructions in the README of this repo. I'll likley not get
to look into this in April, day job is keeping me busy.

**Comment by alexzhang7801** [comment]
I met same issue. Can not running by Eclipse.

**Comment by alexzhang7801** [comment]
Why not set session in the initial part of RS start?

**Comment by alexzhang7801** [comment]
class PXCSession
This class defines a standard interface for maintaining the SDK context. 
This application can query and create instances of I/O and algorithm module
implementations.

**Comment by ghost** [comment]
What's the last SDK that works with lib-gdx? Thanks

**Comment by alexzhang7801** [comment]
Hi Man,
current lib-gdx  version 1.0.0. Thanks.

2015-05-16 23:24 GMT+08:00 Copy_ notifications@github.com:

> What's the last SDK that works with lib-gdx? Thanks
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/libgdx/gdx-realsense/issues/10#issuecomment-102637975
> .

## 

Dear ,

Best Regards,Alex.

alexzhang7801@gmail.com

Mobile Phone:0086-13922850102

**Comment by ghost** [comment]
Sorry, misunderstood me. Which RealSense SDK is reported as working with lib-gdx v1.0? Since the last one doesn't work. thanks

**Comment by ghost** [comment]
For everybody who wants an RSSDK which works with the current lig-gdx here's the link: https://mega.co.nz/#!gAABGbpA!hBDf6aJFCvPy2svQWPJZ9EVdHPMNFYbuizBITbER63o

Thanks to @alexzhang7801 for the help.

**Comment by ghost** [comment]
I can't preform a landmark query points, an exception is thrown. "a heap has been corrupted" and it references this: ntdll.dll

Any guess? Can you help @badlogic ?

thanks

**Comment by alexzhang7801** [comment]
Hi Rafael,
Could you support capture image form eclipse IDE?
thanks

2015-05-18 1:36 GMT+08:00 Copy_ notifications@github.com:

> I can't preform a landmark query points, an exception is thrown. "a heap
> has been corrupted" and it references this: ntdll.dll
> 
> Any guess? Can you help @badlogic https://github.com/badlogic ?
> 
> thanks
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/libgdx/gdx-realsense/issues/10#issuecomment-102825189
> .

## 

Dear ,

Best Regards,Alex.

alexzhang7801@gmail.com

Mobile Phone:0086-13922850102

**Comment by ghost** [comment]
Yes I can see the depth and color image, preform face tracking and pose estimation. But I can't find the landmark points the exception is thrown.

**Comment by ramyalsharif** [comment]
anything new on this?

PXCSenseManager.CreateInstance()

returns null

**Comment by QuocZung** [comment]
I'm having the same problem: PXCSenseManager.CreateInstance() returns always null. I'm a student and pretty new to this so pls patient with me. I installed RealSense SDK 2.0 and its working fine with Viewer, C++ examples but not with C#. Any solutions pls?
